app/javascript/components/button-group/index.jsx

```diff
@@ -0,0 +1,176 @@
+import MiqFormRenderer, { useFormApi } from '@@ddf';
+import { Loading } from 'carbon-components-react';
+import PropTypes from 'prop-types';
+import { FormSpy } from '@data-driven-forms/react-form-renderer';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import createSchema from './group-form.schema';
+
+const GroupForm = ({
+  rec_id, available_fields, fields, url, applies_to_class
+}) => {
+  const [{
+    isLoading, initialValues, buttonIcon, options,
+  }, setState] = useState({
+    isLoading: true,
+  });
+
+  let iconChanged = false;
+
+  /** Function to change the format of the props available_fields & fields from the format
+   * [[string,number]] to [{label:string, value:number}] to use as options in dual-list-select */
+  const formatButton = (buttons) => {
+    const options = [];
+    buttons && buttons.map((btn) => options.push({ label: btn[0], value: btn[1] }));
+    return options;
+  };
+
+  /** available_fields or an empty array is passed as props. */
+  const buttonOptions = formatButton(available_fields.concat(fields));
+
+  /** Function to extract the ButtonGroupName from api result like 'ButtonGroupName | xyz'. */
+  const formatName = (buttonGroupName) => {
+    if(!buttonGroupName){
+      return undefined;
+    }
+    return buttonGroupName.split('|')[0].trim('');
+  }
+
+  /** Function to update the field set_data's values during form submit */
+  const formatSetData = (set_data) => {
+    return {  ...set_data, 
+                button_color: (set_data && set_data.button_color) ? set_data.button_color : '#000', 
+                button_icon: buttonIcon, 
+                applies_to_class: applies_to_class };
+  }
+
+  useEffect(() => {
+    if (rec_id) {
+      API.get(`/api/custom_button_sets/${rec_id}`).then((initialValues) => {
+        setState((state) => ({
+          ...state,
+          initialValues: {...initialValues, name: formatName(initialValues.name)},
+          buttonIcon: (initialValues && initialValues.set_data) ? initialValues.set_data.button_icon : '',
+          options: buttonOptions,
+          isLoading: false,
+        }));
+      });
+    } else {
+      setState((state) => ({
+        ...state,
+        options: available_fields.length > 0 ? buttonOptions : [],
+        isLoading: false,
+      }));
+    }
+  }, []);
+
+  iconChanged = (rec_id && initialValues && initialValues.set_data) 
+                  ? (initialValues.set_data.button_icon !== buttonIcon) 
+                  : false;
+
+  const onSubmit = (values) => {
+    miqSparkleOn();
+    values.set_data = formatSetData(values.set_data);
+    const request = rec_id
+      ? API.patch(`/api/custom_button_sets/${rec_id}`, values)
+      : API.post('/api/custom_button_sets', values);
+    request.then(() => {
+      const message = sprintf(
+        rec_id
+          ? __('Modification of Button group "%s" has been successfully.')
+          : __('Button group "%s" has been successfully added.'),
```

Comment:
This should be `Add Button group "%s" has been successfully queued.`

Reply:
ok